lib/ansible/utils/encrypt.py

```diff
@@ -170,7 +174,10 @@ def _hash(self, secret, salt, salt_size, rounds):
 
         # starting with passlib 1.7 'using' and 'hash' should be used instead of 'encrypt'
         if hasattr(self.crypt_algo, 'hash'):
-            result = self.crypt_algo.using(**settings).hash(secret)
+            if self.algorithm in ('msdcc', 'msdcc2'):
+                result = self.crypt_algo.hash(secret, **settings)
```

Comment:
So, this doesn't verify anymore that user is set, unlike the previous PR, I think that should be corrected.

Reply:
It is done implicitly by passlib.
So if the user is not set an exception would be thrown, similar to what https://github.com/ansible/ansible/issues/45392 experienced.

This is the same behaviour as for all other hashing algorithms.
I.e. the checking of what parameters are supported/needed is done by passlib.

Reply:
Throwing a exception do not seems like great UX for users.

Reply:
I think from a maintenance point it is easier to let passlib handle the parameters.
Otherwise we would have to take care of each algorithm ourselves and have a list of required and optional parameters for them.
It becomes worse if algorithms retrieve new paramters that we already support.
Now iff new algorithms are added to passlib we support them automatically.
Given that they use the 'using' and 'hash' functions.